genai-perf/genai_perf/inputs/retrievers/synthetic_data_retriever.py

```diff
@@ -105,44 +112,49 @@ def _generate_stateless_entries(self, use_prefix_prompts: bool) -> List[DataRow]
         data_rows = []
 
         for _ in range(self.config.num_dataset_entries):
-            row = self._create_data_row(use_prefix_prompts)
-            row.images = [
-                self._generate_image() for _ in range(self.config.batch_size_image)
-            ]
+            row = self._create_data_row()
+            row.texts = self._generate_prompts(use_prefix_prompts)
+            row.images = self._generate_images()
             data_rows.append(row)
 
         return data_rows
 
-    def _create_data_row(
-        self, use_prefix_prompts: bool, session_id: str = ""
-    ) -> DataRow:
-        row = DataRow(texts=[], images=[], optional_data={}, payload_metadata={})
+    def _create_data_row(self, session_id: str = "") -> DataRow:
+        row = DataRow()
 
         if session_id:
             row.payload_metadata["session_id"] = session_id
-
-        row.texts = [
-            self._generate_prompt(use_prefix_prompts)
-            for _ in range(self.config.batch_size_text)
-        ]
         return row
 
-    def _generate_prompt(self, use_prefix_prompts: bool) -> str:
-        prompt = SyntheticPromptGenerator.create_synthetic_prompt(
-            self.config.tokenizer,
-            self.config.prompt_tokens_mean,
-            self.config.prompt_tokens_stddev,
-        )
-        if use_prefix_prompts:
-            prefix_prompt = SyntheticPromptGenerator.get_random_prefix_prompt()
-            return f"{prefix_prompt} {prompt}"
-        return prompt
-
-    def _generate_image(self):
-        return SyntheticImageGenerator.create_synthetic_image(
-            image_width_mean=self.config.image_width_mean,
-            image_width_stddev=self.config.image_width_stddev,
-            image_height_mean=self.config.image_height_mean,
-            image_height_stddev=self.config.image_height_stddev,
-            image_format=self.config.image_format,
-        )
+    def _generate_prompts(self, use_prefix_prompts: bool) -> List[str]:
+        prompts = []
+        for _ in range(self.config.batch_size_text):
```

Comment:
why did you unfold the list comprehension into a for loop?

Reply:
We have multiple places that needs this functionality (since I took it out of `_create_data_row`). It's easier & more readable to call it as a function than list comprehensions.

Reply:
I personally agree that for loops are easier to read. However, list comprehensions are more pythonic, so we should get used to reading them, if only to ensure they are an option for us and we dismiss them out of hand.

If you are using the underlying data in multiple locations, i agree the for loop in a method makes more sense.